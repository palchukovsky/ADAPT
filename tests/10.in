DECLARE testDecl;

SCOPE level1 {

  DECLARE testDecl;

  SCOPE level2 {

    DECLARE testDecl;

    SCOPE level3 { DECLARE testDecl3; }

    USING level3;
    ACCESS testDecl;
    ACCESS testDecl3;

  }

  USING level2::level3;

  ACCESS testDecl;
  ACCESS level2::level3::testDecl3;
  ACCESS testDecl3;

  USING ::level1::level2::level3;
  ACCESS testDecl3;

}

USING level1::level2::level3;

ACCESS testDecl3;
ACCESS level1::level2::level3::testDecl3;
