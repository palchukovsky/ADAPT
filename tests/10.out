LINE 14 ACCESS ::level1::level2::testDecl
LINE 15 ACCESS ::level1::level2::level3::testDecl3
LINE 21 ACCESS ::level1::testDecl
LINE 22 ACCESS ::level1::level2::level3::testDecl3
LINE 23 ACCESS ::level1::level2::level3::testDecl3
LINE 26 ACCESS ::level1::level2::level3::testDecl3
LINE 32 ACCESS ::level1::level2::level3::testDecl3
LINE 33 ACCESS ::level1::level2::level3::testDecl3
